src/shared_gpu_k_clustering.c

```diff
@@ -0,0 +1,50 @@
+#include <float.h>
+#include <stdlib.h>
+
+#include "shared_gpu_k_clustering.h"
+
+extern void cuda_Kernel_1(Point* points_h, int points_size, Point c, int cluster_id);
+extern void cuda_Kernel_2(Point* points_h, int points_size, int* n_points, double* sum_x, double* sum_y, double* sum_z, int k);
+
+// perform the c clustering
+void k_means_clustering(Point* points, int points_size, int epochs, int k) {
+	Point centroids[k];
+	srand(42);
+	for (int i = 0; i < k; ++i) {
+		centroids[i] = points[rand() % points_size];
+	}
+
+	for (int i = 0; i < epochs; ++i) {
+		// For each centroid, compute distance from centroid to each point
+		// and update point's cluster if necessary
+		for (int cluster_id = 0; cluster_id < k; ++cluster_id){
+			Point c = centroids[cluster_id];
+			cuda_Kernel_1(points, points_size, c, cluster_id);
```

Comment:
I wonder if we could move this kernel call above the cluster_id for loop, and have the kernel compute the cluster_id